apps/src/sites/studio/pages/signup.js

```diff
@@ -38,22 +41,33 @@ window.SignupManager = function (options) {
       "school_info.zip"
     ];
 
-    for (var i = 0; i < fields.length; i++) {
-      var field = fields[i];
-      if (err.responseJSON.errors[field]) {
-        var errorField = $(`#${field}-block .error_in_field`);
-         // We have a custom inline message for user_type errors already set in the DOM.
-        if (field === "terms_of_service_version") {
-          errorField.text(self.options.acceptTermsString);
-        } else if (field === "school_info.zip") {
-          errorField = $('#school-zip').find('.error_in_field');
-          errorField.text(err.responseJSON.errors[field][0]);
-        } else if (field !== "user_type") {
-          errorField.text(err.responseJSON.errors[field][0]);
+    var fieldsWithErrors = 0;
+    if (err.responseJSON && err.responseJSON.errors) {
+      for (var i = 0; i < fields.length; i++) {
+        var field = fields[i];
+        if (err.responseJSON.errors[field]) {
+          var errorField = $(`#${field}-block .error_in_field`);
+          // We have a custom inline message for user_type errors already set in the DOM.
+          if (field === "terms_of_service_version") {
+            errorField.text(self.options.acceptTermsString);
+          } else if (field === "school_info.zip") {
+            errorField = $('#school-zip').find('.error_in_field');
+            errorField.text(err.responseJSON.errors[field][0]);
+          } else if (field !== "user_type") {
+            errorField.text(err.responseJSON.errors[field][0]);
+          }
+          errorField.fadeTo("normal", 1);
+          fieldsWithErrors++;
         }
-        errorField.fadeTo("normal", 1);
       }
     }
+
+    // if we did not receive a response that had field-specific error information,
+    // show a generic error
+    if (fieldsWithErrors === 0) {
+      $('#signup-error').show();
+      console.log("got unhandled error");
```

Comment:
should this be console.error instead of console.log? Is there anything more useful we could/should but as the text?

Reply:
Thanks! That `console.log` was for my own debugging and meant to remove, missed it. Good catch. Removed.